src/hooks/NetceteraThreeDsHooks.res

```diff
@@ -0,0 +1,443 @@
+        challengeParams.acsTransactionId,
+        challengeParams.threeDSRequestorURL,
+        challengeParams.threeDSServerTransId,
+        status => {
+          logger(
+            ~logType=INFO,
+            ~value=status->JSON.stringifyAny->Option.getOr(""),
+            ~category=USER_EVENT,
+            ~eventName=NETCETERA_SDK,
+            (),
+          )
+          status->isStatusSuccess
+            ? Netcetera3dsModule.generateChallenge(status => {
+                logger(
+                  ~logType=INFO,
+                  ~value=status->JSON.stringifyAny->Option.getOr(""),
+                  ~category=USER_EVENT,
+                  ~eventName=NETCETERA_SDK,
+                  (),
+                )
+                switch (status.status, status.message) {
+                | ("success", _) => {
+                    let authoriseUrl = threeDsData.threeDsAuthorizeUrl
+                    hsAuthoriseCall(~authoriseUrl, ~onSuccess, ~onFailure)->ignore
+                  }
+                | (_, _) =>
+                  shortPollStatusAndRetrieve(~pollConfig=threeDsData.pollConfig, ~publishableKey)
+                }
+              })
+            : onFailure(threeDsSdkChallengeStatus.errorMsg)
+        },
+      )
+    }
+
+    let frictionlessAuthroiseAndContinue = (
+      ~threeDsData: PaymentConfirmTypes.threeDsData,
+      ~onSuccess: string => unit,
+      ~onFailure: string => unit,
+    ) => {
+      hsAuthoriseCall(~authoriseUrl=threeDsData.threeDsAuthorizeUrl, ~onSuccess, ~onFailure)->ignore
+    }
+
+    let hsThreeDsAuthCall = (
+      clientSecret: string,
+      publishableKey: string,
+      aReqParams: aReqParams,
+      threeDsData: PaymentConfirmTypes.threeDsData,
+      onSuccess: string => unit,
+      onFailure: string => unit,
+    ) => {
+      let uri = threeDsData.threeDsAuthenticationUrl
+      let bodyStr = generateAuthenticationCallBody(clientSecret, aReqParams)
+      let headers = getAuthCallHeaders(publishableKey)
+
+      apiLogWrapper(
+        ~logType=INFO,
+        ~eventName=AUTHENTICATION_CALL_INIT,
+        ~url=uri,
+        ~statusCode="",
+        ~apiLogType=Request,
+        ~data=JSON.Encode.null,
+        (),
+      )
+
+      CommonHooks.fetchApi(~uri, ~bodyStr, ~headers, ~method_=Post, ())
+      ->Promise.then(data => {
+        let statusCode = data->Fetch.Response.status->string_of_int
+        if statusCode->String.charAt(0) === "2" {
+          data
+          ->Fetch.Response.json
+          ->Promise.then(res => {
+            apiLogWrapper(
+              ~logType=INFO,
+              ~eventName=AUTHENTICATION_CALL,
+              ~url=uri,
+              ~statusCode,
+              ~apiLogType=Response,
+              ~data=JSON.Encode.null,
+              (),
+            )
+            let authResponse = res->authResponseItemToObjMapper
+
+            switch authResponse {
+            | AUTH_RESPONSE(challengeParams) =>
+              logger(
```

Comment:
move this log outside the switch case. We would want to see the value even if its not "C"